server/app/services/agent/node_plugger.rb

```diff
@@ -29,7 +31,15 @@ def plugin!
     end
 
     def update_node
-      node.set(connected: true, last_seen_at: Time.now.utc)
+      result = HostNode.where(:id => node.id).any_of(
+        {:connected_at => nil}, {:connected_at.lt => @connected_at}
+      ).update(connected: true, last_seen_at: Time.now.utc, connected_at: @connected_at)
```

Comment:
Depends on what you want to do. :) AFAIK this will fire first the query and after that the update. If you want that to be one single operation you'd have to use `find_and_modify`.

I'm not sure how this behaves now when `connected_at` is a string. Feels a bit fishy though. :)